sled-storage/src/manager.rs

```diff
@@ -815,6 +992,131 @@ impl StorageManager {
         }
     }
 
+    // Lists datasets that this zpool contains
+    async fn datasets_list(
+        &self,
+        zpool: &ZpoolName,
+    ) -> Result<Vec<DatasetProperties>, Error> {
+        let log = self.log.new(o!("request" => "datasets_list"));
+
+        let datasets_of_interest = [
+            // We care about the zpool itself, and all direct children.
+            zpool.to_string(),
+            // Likewise, we care about the encrypted dataset, and all
+            // direct children.
+            format!("{zpool}/{CRYPT_DATASET}"),
+            // The zone dataset gives us additional context on "what zones
+            // have datasets provisioned".
+            format!("{zpool}/{ZONE_DATASET}"),
+        ];
+
+        info!(log, "Listing datasets within zpool"; "zpool" => zpool.to_string());
+        illumos_utils::zfs::Zfs::get_dataset_properties(
```

Comment:
dumb style nitpick: `Zfs` is imported...
```suggestion
        Zfs::get_dataset_properties(
```

Reply:
Patched